Sources_v3/StreamChatUI/MessageActionsPopup/ChatMessagePopupViewController.swift

```diff
@@ -0,0 +1,113 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import UIKit
+
+open class ChatMessagePopupViewController<ExtraData: UIExtraDataTypes>: ViewController, UIConfigProvider {
+    public private(set) lazy var scrollView = UIScrollView()
+        .withoutAutoresizingMaskConstraints
+    public private(set) lazy var scrollContentView = UIView()
+        .withoutAutoresizingMaskConstraints
+    public private(set) lazy var blurView = UIVisualEffectView(effect: UIBlurEffect(style: .light))
+        .withoutAutoresizingMaskConstraints
+    public private(set) lazy var messageContentView = uiConfig.messageList.messageContentView.init()
+        .withoutAutoresizingMaskConstraints
+
+    public let message: _ChatMessageGroupPart<ExtraData>
+    public let messageViewFrame: CGRect
+    public let actionsController: ChatMessageActionsViewController<ExtraData>
+    public let reactionsController: ChatMessageReactionViewController<ExtraData>
+
+    // MARK: - Init
+
+    public init(
+        message: _ChatMessageGroupPart<ExtraData>,
+        messageFrameToFit: CGRect,
+        messageActionsController: ChatMessageActionsViewController<ExtraData>,
+        messageReactionsController: ChatMessageReactionViewController<ExtraData>
+    ) {
+        self.message = message
+        messageViewFrame = messageFrameToFit
+        actionsController = messageActionsController
+        reactionsController = messageReactionsController
+        
+        super.init(nibName: nil, bundle: nil)
+    }
```

Comment:
Interesting that you chose this pattern for instantiating the view controller? Why not having the VC to create the subviews from UIConfig?

Reply:
The subviews that can be overridden are owned by `messageActionsController/messageReactionsController` and instantiated from `UIConfig`

Or.. I didn't get your point 😄

Reply:
Now when `ChatMessagePopupViewController` is a view config provider, it should initialize the subviews via `UIConfig`. We shouldn't override the initializer and set the dependencies (message and frame) the same way we set controllers for example.

Reply:
removed `init` and set the dependencies after the controller is created in 35afc5747712e70c838c2a6af65527ed985e46d6